client/my-sites/upgrades/domain-management/dns/dns-add-new.jsx

```diff
@@ -107,14 +104,16 @@ const DnsAddNew = React.createClass( {
 	},
 
 	onChange( event ) {
+		const { name, value } = event.target;
 		this.formStateController.handleFieldChange( {
-			name: event.target.name,
-			value: event.target.value.trim().toLowerCase()
+			name,
+			value: name === 'data' && this.state.type === 'TXT' ? value : value.trim().toLowerCase(),
```

Comment:
It might make sense to move this logic to a function or a variable something like `shouldNormalize`.